src/components/UserCard.js

```diff
@@ -21,17 +22,19 @@ export default class UserCard extends Component {
         })
     }
 
-    handleSearch=async (text)=>{
-        let response=await axios.get(`https://api.github.com/users/${text}`);
-        let data=response.data;
+    handleSearch=(text)=>{
+        this.setState({
+            userName:text,
+        })
+    }
 
-        if(data == undefined){
-            return;
-        }
+    handleSearching= async ()=>{
+        let response=await axios.get(`https://api.github.com/users/${this.state.userName}`);
+        let data=response.data;
```

Comment:
create a util for api hit and handle all the success and error states over there